tests/src/test/scala/org/http4s/MethodSpec.scala

```diff
@@ -26,10 +29,33 @@ class MethodSpec extends Http4sSpec {
 
   "only tokens are valid methods" in {
     prop { (s: String) =>
+      // TODO: this check looks meaningless: `fromString` mostly relies on `Rfc2616BasicRules.token`
+      //       and the result is compared to `isToken` which calls to `Rfc2616BasicRules.token` as well.
```

Comment:
Yeah, it's difficult to avoid repeating implementation in some property tests.  These kind of properties guard against future optimizations that break the contract, but that's unlikely here.